packages/sdk-react/src/ui/CounterpartCell/CounterpartCell.tsx

```diff
@@ -1,77 +1,69 @@
-import { useCallback } from 'react';
-
 import { components } from '@/api';
+import { getCounterpartName } from '@/components/counterparts/helpers';
 import { useCounterpartById } from '@/core/queries';
-import CorporateFareIcon from '@mui/icons-material/CorporateFare';
-import PersonIcon from '@mui/icons-material/Person';
-import { Chip, Box, Avatar, Skeleton, Typography } from '@mui/material';
+import { Avatar, Box, Chip, Skeleton, Typography } from '@mui/material';
 
 interface CounterpartCellProps {
   counterpartId: components['schemas']['CounterpartResponse']['id'];
 }
 
-export const CounterpartCell = ({ counterpartId }: CounterpartCellProps) => {
-  const { data: counterpart, isLoading } = useCounterpartById(counterpartId);
-
-  const getCounterpartText = useCallback(
-    (counterpart: components['schemas']['CounterpartResponse']) => {
-      return counterpart.type === 'organization'
-        ? (
-            counterpart as components['schemas']['CounterpartOrganizationRootResponse']
-          ).organization.legal_name
-        : `${
-            (
-              counterpart as components['schemas']['CounterpartIndividualRootResponse']
-            ).individual.first_name
-          } ${
-            (
-              counterpart as components['schemas']['CounterpartIndividualRootResponse']
-            ).individual.last_name
-          }`;
-    },
-    []
-  );
-
-  if (!counterpartId || (!isLoading && !counterpart)) {
-    return null;
+const calculateColorIndex = (name: string) => {
+  let sum = 0;
+  for (let i = name.length - 1; i >= 0; i--) {
+    sum += name.charCodeAt(i);
   }
+  return sum % 5;
+};
 
+export const CounterPartCellByName = ({
+  name,
+  isLoading,
+}: {
+  name: string;
+  isLoading?: boolean;
+}) => {
+  if (!name) return null;
+  const nameParts = name.split(' ');
+  const avatarLetters = (
+    nameParts.length >= 2
+      ? nameParts[0][0] + nameParts[nameParts.length - 1][0]
+      : name[0] || ''
```

Comment:
Some of the logic here looks like a magic numbers to me as has not dive deep. What you think to provide some extra comments here or reduce complexity?

Maybe something like:
`nameParts.length >= 2: // Checks if there are at least two parts in the nameParts array.
`
...